source/melpomene/src/sim_drivers/emb_display.rs

```diff
@@ -268,48 +261,53 @@ impl DisplayInfo {
         width: u32,
         height: u32,
     ) -> Result<FrameChunk, FrameError> {
-        let found = self.frames.iter_mut().find(|f| f.is_none());
-
-        if let Some(slot) = found {
-            let fidx = self.frame_idx;
-            self.frame_idx = self.frame_idx.wrapping_add(1);
-
-            *slot = Some(FrameInfo { frame: fidx });
+        let fidx = self.frame_idx;
+        self.frame_idx = self.frame_idx.wrapping_add(1);
 
-            let size = (width * height) as usize;
+        match self.frames.try_push(FrameInfo { frame: fidx }) {
+            Ok(_) => {}
+            Err(_) => {
+                return Err(FrameError::NoFrameAvailable);
+            }
+        }
```

Comment:
tiny unimportant nit: could just be 
```suggestion
        self
            .frames.try_push(FrameInfo { frame: fidx })
            .map_err(|_| FrameError::NoFrameAvailable)?;
```